types/js-cookie/index.d.ts

```diff
@@ -88,10 +88,11 @@ declare namespace Cookies {
          * will run the converter first for each cookie. The returned
          * string will be used as the cookie value.
          */
-        withConverter(converter: CookieConverter | { write: CookieConverter; read: CookieConverter; }): CookiesStatic;
+        withConverter(converter: CookieReadConverter | { write: CookieWriteConverter; read: CookieReadConverter; }): CookiesStatic;
     }
 
-    type CookieConverter = (value: string, name: string) => string;
+    type CookieWriteConverter = (value: string | object, name: string) => string;
```

Comment:
You shouldn't use `object` here. If you do, users always will have to convert this to an object with an index signature: `{ [key: string]: any; }` to access properties (or something more specific). It is better if you type it as an object with a type signature directly to spare your users this unnecessary conversion. And I suspect that the values aren't really `any` typed, I suppose they are `string`s or `string[]` or something like this.

Reply:
I used `object` since this reflected the type used in the `set` method. 

Since the value is passed directly from the argument in the `set`method to the `CookieWriteConverter`, I felt the two should have the same type.

Reply:
Ah, OK. Then you should do the following:
```typescript
interface CookiesStatic<T extends object = object> {
  // ...
  set(name: string, value: string | T, options?: CookieAttributes): void;
  // ...
  withConverter(converter: CookieReadConverter | { write: CookieWriteConverter<T>; read: CookieReadConverter; }): CookiesStatic;
}

type CookieWriteConverter<T extends object> = (value: string | T, name: string) => string;
```

Reply:
Sorry, my original version was incorrect. Typed it down without testing.

Reply:
```typescript
interface CookiesStatic<T extends object = object> {
  // ...
  set(name: string, value: string | T, options?: CookieAttributes): void;
  // ...
  withConverter<TConv extends object>(
    converter:
      | CookieReadConverter
      | { write: CookieWriteConverter<TConv>; read: CookieReadConverter }
  ): CookiesStatic<TConv>;
}

type CookieWriteConverter<T extends object> = (value: string | T, name: string) => string;
```